scripts/dockerfiles/Dockerfile.init

```diff
@@ -2,7 +2,7 @@ ARG golang_image
 
 FROM $golang_image as builder
```

Comment:
I think the build will go faster if we set `FROM --platform=$BUILDPLATFORM as builder` here.

(That means we'll cross-compile on the native architecture, rather compile in a qemu-emulated version of the target architecture)

Ditto for other Dockerfiles.

Reply:
Good catch, yes, will change this part. 
https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/ has this recommendation too for faster multi-platform builds.

Reply:
Usage of BUILDPLATFORM in docker file for instruction `FROM --platform=$BUILDPLATFORM $golang_image as builder` has the same issue as TARGETARCH when buildkit isn't used, as it is a platform argument available with [buildkit](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope) only.

```
Step 2/17 : FROM --platform=$BUILDPLATFORM $golang_image as builder
failed to parse platform : "" is an invalid component of "": platform specifier component must match "^[A-Za-z0-9_-]+$": invalid argument
```

We could mandate usage of buildkit by [setting](https://docs.docker.com/develop/develop-images/build_enhancements/#to-enable-buildkit-builds) `DOCKER_BUILDKIT=1` in Makefile prior to docker build. 
In that way, we would get the platform arguments and simplify the Makefile and obtain faster build speeds.
Thoughts on mandating the buildkit setting ?

Reply:
After setting `DOCKER_BUILDKIT=1`, if we execute go build using `BUILDPLATFORM`, we would have to pass additional arguments to `build-linux` target (to set the GOARCH value during go build).

However, such approach does mean that execution of `make build-linux` without docker commands would not work without manually / automatically setting GOARCH in the makefile in that invocation path (which would revert makefile to the approach in previous revision). Thus, it may be better to leave the dockerfiles as-is for simpler makefile logic. Thoughts ?

Reply:
> Thoughts?

I think trying to support builds using both buildx and older docker from the same Dockerfile is making everything complicated and messy.

Do we need to continue to support building with older docker?  If so, maybe we don't want to support buildx yet?

(This is a build-time dependency only, so we can be opinionated.)

Reply:
So, with the makefile change, it will only mandate usage of docker buildkit which has been available in docker since version 18.09 (https://docs.docker.com/develop/develop-images/build_enhancements/).

Usage of docker buildx cli within the existing makefile commands isn't needed and the usual docker cli can be used with docker buildkit too.